src/components/dialogs/field-hook.js

```diff
@@ -308,56 +323,163 @@ export const useEquipmentTableValues = ({
         checkValues();
     }, [checkValues]);
-
-    const handleDeleteItem = useCallback(
-        (index) => {
-            setValues((oldValues) => {
-                let newValues = [...oldValues];
+    const [openCSVImportDialog, setOpenCSVImportDialog] = useState(false);
+    const [selectedIds, setSelectedIds] = useState(new Set());
+    const handleDeleteItem = useCallback(() => {
+        setValues((oldValues) => {
+            let newValues = [...oldValues];
+            if (selectedIds.size === newValues.length) {
+                setSelectedIds(new Set());
+                return [{}];
+            }
+            selectedIds.forEach((index) => {
                 newValues.splice(index, 1);
-                setIsEdited(true);
-                return newValues.length === 0 ? [{}] : newValues;
             });
-            setCreateFilterErr('');
-        },
-        [setCreateFilterErr, setIsEdited]
-    );
+            setSelectedIds(new Set());
+            setIsEdited(true);
+            return newValues.length === 0 ? [{}] : newValues;
+        });
+        setCreateFilterErr('');
+    }, [selectedIds, setCreateFilterErr, setIsEdited]);
 
     const handleSetValue = useCallback(
         (index, newValue) => {
             setValues((oldValues) => {
                 let newValues = [...oldValues];
                 newValues[index] = newValue;
-                setIsEdited(true);
                 return newValues;
             });
             setCreateFilterErr('');
+            setIsEdited(true);
         },
         [setCreateFilterErr, setIsEdited]
     );
 
     const handleChangeOrder = useCallback(
-        (index, direction) => {
+        (direction) => {
             const res = [...values];
-            const [item] = res.splice(index, 1);
-            res.splice(index + direction, 0, item);
+            const result = [...selectedIds];
+            result.sort();
+
+            let isContiguous = true;
+            for (let i = 0; i < result.length - 1; i++) {
+                if (result[i + 1] - result[i] !== 1) {
+                    isContiguous = false;
+                    break;
+                }
+            }
+
+            if (isContiguous) {
+                if (direction === -1) {
+                    const [item] = res.splice(result[0] - 1, 1);
+                    res.splice(result[result.length - 1], 0, item);
+                } else {
+                    const [item] = res.splice(result[result.length - 1] + 1, 1);
+                    res.splice(result[0], 0, item);
+                }
+            } else {
+                selectedIds.forEach((elem) => {
+                    const [item] = res.splice(elem, 1);
+                    res.splice(elem + direction, 0, item);
+                });
+            }
+            let array = [...selectedIds];
+            array.sort();
+            for (let i = 0; i < array.length; i++) {
+                array[i] = array[i] + direction;
+            }
```

Comment:
this can be like this: 
const array = Array.from(selectedIds)
                .sort()
                .map((val) => val + direction);